### ISSUE TITLE
Remove use of static context from HabitsApplication class

### ISSUE THREAD

**Issue by AnirudhaAgashe** [iSoron/uhabits#222]
Holding a static reference to the context in application can lead to memory leaks.

Also the call back to [onTerminate ](https://developer.android.com/reference/android/app/Application.html#onTerminate())is only received in case of emulated environment and not on actual device.

**Comment by AnirudhaAgashe** [comment]
I was able to replace the static context in most of the classes without any issues by injecting Application and activity context wherever applicable. In case PebbleReceiver used the context that is received in receiveData call back. Also implemented a alternative method of checking isTestMode which doesn't required context.

However in case of ListHabitsController I had to inject activity context. ListHabitsController requires context to carry out ExportCSVTask and ExportDBTask.

Is there any idea how we can inject context for ExportCSVTask and ExportDBTask without having to inject context in ListHabitsController ?
